birdyboard.py

```diff
@@ -0,0 +1,505 @@
+import pickle
+
+
+class Birdyboard:
+
+    def __init__(self):
+        self.users = []
+        self.user_name = None
+
+        self.chirps_library = {"public": [], "private": []}
+        # these variables are assigned when a specific chirp thread is being viewed or added to.
+        self.public_or_private = ""
+        self.chirp_index = None
+
+# ###############################
+# ######## MENU PRINTERS ########
+# ###############################
+
+        self.unlogged_in_menu = "Welcome. You are not logged in.\n1. new user\n2. log in\n3. view public chirps\n'x' to exit.\n"
+        self.logged_in_menu = "Welcome " + self.user_name + "!\n1. log out\n2. view public chirps\n3. view private chirps\n4. New public chirp thread\n5. new private chirp thread\n'x' to exit.\n"
```

Comment:
good reason to put it in a function! ;)

On Wed, Aug 3, 2016 at 10:42 AM, Ryan Tanay <notifications@github.com>
wrote:

> In birdyboard.py
> <https://github.com/nss-day-cohort-13/birdyboard-MeganCFord/pull/3#discussion_r73352298>
> :
>
> > +
> > +    def __init__(self):
> > +        self.users = []
> > +        self.user_name = None
> > +
> > +        self.chirps_library = {"public": [], "private": []}
> > +        # these variables are assigned when a specific chirp thread is being viewed or added to.
> > +        self.public_or_private = ""
> > +        self.chirp_index = None
> > +
> > +# ###############################
> > +# ######## MENU PRINTERS ########
> > +# ###############################
> > +
> > +        self.unlogged_in_menu = "Welcome. You are not logged in.\n1. new user\n2. log in\n3. view public chirps\n'x' to exit.\n"
> > +        self.logged_in_menu = "Welcome " + self.user_name + "!\n1. log out\n2. view public chirps\n3. view private chirps\n4. New public chirp thread\n5. new private chirp thread\n'x' to exit.\n"
>
> This throws a hard error and crashes out both your app and your tests, as
> setting the value of self.logged_in_menu requires converting
> self.user_name to a string, and at this point in the program flow,
> self.user_name is None and the NoneType can't be converted to a string.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nss-day-cohort-13/birdyboard-MeganCFord/pull/3/files/506b935ffbc8e2f3a3bd8c9c47fabec18a78913b#r73352298>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARUAVY7VGSk-_5b-SOgUBPmUI1j8jDoAks5qcLbOgaJpZM4JaAC3>
> .
>